statsd/statsd.go

```diff
@@ -235,24 +256,25 @@ func (c *Client) joinMaxSize(cmds [][]byte, sep string, maxSize int) ([][]byte,
 			needed = needed + sepLen
 		}
 
-		if c.buffer.Len()+needed <= maxSize {
+		if len(buffer)+needed <= maxSize {
 			if elem != 0 {
-				c.buffer.Write(sepBytes)
+				buffer = append(buffer, sepBytes...)
 			}
-			c.buffer.Write(cmd)
+			buffer = append(buffer, cmd...)
 			elem++
 		} else {
-			frames = append(frames, copyAndResetBuffer(&c.buffer))
+			frames = append(frames, buffer)
 			ncmds = append(ncmds, elem)
 			// if cmd is bigger than maxSize it will get flushed on next loop
-			c.buffer.Write(cmd)
+			buffer = make([]byte, 0, maxSize)
+			buffer = append(buffer, cmd...)
```

Comment:
this will be new allocation. just buffer = buffer[:0] will reuse the same allocation.